src/staking/manager.ts

```diff
@@ -0,0 +1,660 @@
+    stakingParamVersion: number,
+    stakingTx: Transaction,
+    feeRate: number,
+  ): Promise<Transaction> {
+    const p = getBbnParamByVersion(stakingParamVersion, this.stakingParams);
+    if (!p) {
+      throw new Error(`Unable to find staking params for version ${stakingParamVersion}`);
+    }
+
+    const stakerInfo = await this.btcProvider.getStakerInfo();
+    if (!stakerInfo) {
+      throw new Error("Staker info not found while creating signed unbonding transaction");
+    }
+
+    const staking = new Staking(
+      this.network,
+      stakerInfo,
+      p,
+      stakingInput.finalityProviderPkNoCoordHex,
+      stakingInput.stakingTimelock,
+    );
+
+    const { psbt } = staking.createWithdrawStakingExpiredTransaction(
+      stakingTx,
+      feeRate,
+    );
+
+    const signedWithdrawalPsbtHex = await this.btcProvider.signPsbt(
+      psbt.toHex()
+    );
+    return Psbt.fromHex(signedWithdrawalPsbtHex).extractTransaction();
+  }
+
+  /**
+   * Creates a signed withdrawal transaction on the slashing path that is 
+   * ready to be sent to the BTC network.
+   * @param stakingInput - The staking inputs.
+   * @param stakingParamVersion - The param version that was used to create the EOI
+   * @param slashingTx - The slashing transaction.
+   * @param feeRate - The fee rate in satoshis per byte.
+   * @returns The signed withdrawal transaction.
+   */
+  async createWithdrawSlashingBtcTransaction(
+    stakingInput: StakingInputs,
+    stakingParamVersion: number,
+    slashingTx: Transaction,
+    feeRate: number,
+  ): Promise<Transaction> {
+    const p = getBbnParamByVersion(stakingParamVersion, this.stakingParams);
+    if (!p) {
+      throw new Error(`Unable to find staking params for version ${stakingParamVersion}`);
+    }
+
+    const stakerInfo = await this.btcProvider.getStakerInfo();
+    if (!stakerInfo) {
+      throw new Error("Staker info not found while creating withdraw slashing transaction");
+    }
+
+    const staking = new Staking(
+      this.network,
+      stakerInfo,
+      p,
+      stakingInput.finalityProviderPkNoCoordHex,
+      stakingInput.stakingTimelock,
+    );
+
+    const { psbt } = staking.createWithdrawSlashingTransaction(
+      slashingTx,
+      feeRate,
+    );
+
+    const signedSlashingPsbtHex = await this.btcProvider.signPsbt(
+      psbt.toHex()
+    );
+    return Psbt.fromHex(signedSlashingPsbtHex).extractTransaction();
+  }
+
+  private async createBtcDelegationMsg(
+    stakingInstance: Staking,
+    stakingInput: StakingInputs,
+    stakingTx: Transaction,
+    bech32Address: string,
+    stakerInfo: StakerInfo,
+    param: StakingParams,
+    inclusionProof?: btcstaking.InclusionProof,
+  ) {
+    const { transaction: unbondingTx } =
+      stakingInstance.createUnbondingTransaction(stakingTx);
+
+    // Create slashing transactions and extract signatures
+    const { psbt: slashingPsbt } =
+      stakingInstance.createStakingOutputSlashingTransaction(stakingTx);
```

Comment:
A bit weird that some functions with the `Transaction` suffix are returning transactions, while others psbts.